src/Core/Components/CounterBadge/FluentCounterBadge.razor

```diff
@@ -1,24 +1,34 @@
 ﻿@namespace Microsoft.FluentUI.AspNetCore.Components
 @using Microsoft.FluentUI.AspNetCore.Components.Extensions
 @inherits FluentBadge
-<fluent-counter-badge id="@Id"
-                      class="@ClassValue"
-                      style="@StyleValue"
-                      appearance="@Appearance.ToAttributeValue()"
-                      color="@Color.ToAttributeValue()"
-                      shape="@Shape.ToAttributeValue()"
-                      size="@Size.ToAttributeValue()"
-                      count="@GetCount()"
-                      overflow-count="@OverflowCount"
-                      show-zero="@ShowZero"
-                      dot="@Dot"
-                      @attributes="@AdditionalAttributes">
-    @if (IconStart != null)
-    {
-        <FluentIcon Value="@IconStart" Slot="start" Color="Components.Color.Custom" CustomColor="@GetIconColor()" aria-label="@IconLabel" />
-    }
-    @if (IconEnd != null)
-    {
-        <FluentIcon Value="@IconEnd" Slot="end" Color="Components.Color.Custom" CustomColor="@GetIconColor()" aria-label="@IconLabel" />
-    }
-</fluent-counter-badge>
+
+<AddTag Name="span" TagWhen="@(() => _isAttached)" style="position: relative; display: inline-block;">
```

Comment:
You could simplify this line, creating a new "tag". So the style can be included in the .css file :-)

```suggestion
<AddTag Name="fluent-badge-container" TagWhen="@(() => _isAttached)">
```